packages/dashboard/src/components/widgets/primitives/input/hooks/useWritableSiteWiseProperty.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  DescribeAssetCommand,
+  IoTSiteWiseClient,
+  ListTagsForResourceCommand,
+  ErrorDetails,
+} from '@aws-sdk/client-iotsitewise';
+import { useContext } from 'react';
+import { ClientContext } from '~/components/dashboard/clientContext';
+
+export const WRITABLE_TAG = 'iot-app-kit:writable';
+
+const useWritableSiteWiseProperty = () => {
```

Comment:
Non-blocking: Decompose into sub-routines.

Reply:
Could you help me undertand what you mean by sub-routines here?